src/core/drive/morph_renderer.js

```diff
@@ -0,0 +1,88 @@
+import Idiomorph from "idiomorph"
+import { nextAnimationFrame } from "../../util"
+import { Renderer } from "../renderer"
+
+export class MorphRenderer extends Renderer {
+  async render() {
+    if (this.willRender) await this.#morphBody()
+  }
+
+  get renderMethod() {
+    return "morph"
+  }
+
+  // Private
+
+  async #morphBody() {
+    this.#morphElements(this.currentElement, this.newElement)
+    this.#reloadRemoteFrames()
+
+    this.#dispatchEvent("turbo:morph", { currentElement: this.currentElement, newElement: this.newElement })
```

Comment:
What is the significance of introducing new `turbo:morph` and `turbo:before-frame-morph` events as an alternative to extending existing events like `turbo:render`, `turbo:before-frame-render`, and `turbo:frame-render`? Could we change the shape of those events' details to incorporate information about the `refresh-method` setting being used?

Reply:
@jorgemanrubia this PR is really shaping up!

I think idea of introducing new events instead of extending existing events is the only remaining concept I'm curious to learn more about.

Reply:
@seanpdoyle I think having different event types make it easier to target only the actions you're interested on.